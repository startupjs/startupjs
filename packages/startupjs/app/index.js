module.exports = require('@startupjs/app')
