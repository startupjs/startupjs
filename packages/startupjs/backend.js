export { default } from '@startupjs/backend'
export * from '@startupjs/backend'
