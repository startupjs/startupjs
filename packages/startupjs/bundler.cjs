throw new Error(`
[startupjs/bundler] ERROR! StartupJS bundler api has changed.

Now instead of using general \`require('startupjs/bundler.cjs')\` you have to require
a specific file you need.

Here is what you need to change:

1. \`require('startupjs/bundler.cjs').webpackWebConfig\`
    v v v
   \`require('startupjs/bundler/webpack.web.config.cjs')\`

2. \`require('startupjs/bundler.cjs').webpackServerConfig\`
    v v v
   \`require('startupjs/bundler/webpack.server.config.cjs')\`

3. \`require('startupjs/bundler.cjs').babelConfig\`
    v v v
   \`require('startupjs/bundler/babel.config.cjs')\`

4. \`require('startupjs/bundler.cjs').metroConfig\`
    v v v
   \`require('startupjs/bundler/metro.config.cjs')\`

5. \`require('startupjs/bundler.cjs').rnConfig\`
    v v v
   \`require('startupjs/bundler/react-native.config.cjs')\`

6. \`require('startupjs/bundler.cjs').viteConfig\`
    v v v
   \`require('startupjs/bundler/vite.config.cjs')\`

Additionally, if you did override .js/.jsx compilation in your webpack.web.config.cjs, you
might have used \`getJsxRule\` function there. In which case you should now get it from helpers:

7. \`const { getJsxRule } = require('startupjs/bundler.cjs')\`
    v v v
   \`const { getJsxRule } = require('startupjs/bundler/helpers.cjs')\`
`)
