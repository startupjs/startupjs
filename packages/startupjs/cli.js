#!/usr/bin/env node
'use strict'

import cli from '@startupjs/cli'
import { fileURLToPath } from 'url'
import path, { dirname } from 'path'

const __filename = fileURLToPath(import.meta.url)
const __dirname = dirname(__filename)

const templatesPath = path.join(__dirname, 'templates')

cli.run({ templatesPath })
