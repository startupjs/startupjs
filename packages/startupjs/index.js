export * from '@startupjs/react-sharedb'
export * from '@startupjs/hooks'
// HINT: `isomorphic` means that the code can be executed both
//        on the server and on the client
export * from '@startupjs/isomorphic-helpers'
// dummy babel macro functions for @startupjs/babel-plugin-rn-stylename-inline
export function css () {}
export function styl () {}
export { default as t } from '@startupjs/i18n/client/t.js'
