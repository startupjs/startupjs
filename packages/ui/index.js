// IMPORTANT: This file exports should match the logic of the script
//            '/scripts/babel/createModuleMap.js', which auto-generates
//            full module import paths for tree-shaking to be used by
//            the 'babel-plugin-startupjs'
//
export { default as Alert } from './components/Alert'
export { default as AutoSuggest } from './components/AutoSuggest'
export { default as Avatar } from './components/Avatar'
export { default as Badge } from './components/Badge'
export { default as Br } from './components/Br'
export { default as Breadcrumbs } from './components/Breadcrumbs'
export { default as Button } from './components/Button'
export { default as Card } from './components/Card'
export { default as Carousel } from './components/Carousel'
export { default as Collapse } from './components/Collapse'
export { default as Content } from './components/Content'
export { default as Div } from './components/Div'
export { default as Divider } from './components/Divider'
export * from './components/draggable'
export { default as DrawerSidebar } from './components/DrawerSidebar'
export * from './components/forms'
export { default as Icon } from './components/Icon'
export { default as Layout } from './components/Layout'
export { default as Link } from './components/Link'
export { default as Loader } from './components/Loader'
export { default as Menu } from './components/Menu'
export { default as Modal } from './components/Modal'
export { default as Pagination } from './components/Pagination'
export * from './components/popups'
export { default as Portal } from './components/Portal'
export { default as Progress } from './components/Progress'
export { default as Rating } from './components/Rating'
export { default as Row } from './components/Row'
export { default as Sidebar } from './components/Sidebar'
export { default as SmartSidebar } from './components/SmartSidebar'
export * from './components/table'
export { default as Tabs } from './components/Tabs'
export { default as Tag } from './components/Tag'
export { ToastProvider, ToastComponent as Toast, toast } from './components/toast'
export { default as Tooltip } from './components/Tooltip'
export * from './components/typography'
export { default as User } from './components/User'
export { default as themed } from './theming/themed'
export { default as ThemeProvider } from './theming/ThemeProvider'
export { default as ThemeContext } from './theming/ThemeContext'
export { useMedia } from './hooks'
export { default as uiAppPlugin } from './uiAppPlugin'
export * from './dialogs'
