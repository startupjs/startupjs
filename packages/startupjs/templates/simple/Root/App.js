import { BASE_URL } from '@env'
import init from 'startupjs/init'
import orm from '../model'
import React, { useMemo } from 'react'
import { observer, useModel, useSession, useDoc } from 'startupjs'
import { TestComponent } from 'components'

// Init startupjs connection to server and the ORM.
// baseUrl option is required for the native to work - it's used
// to init the websocket connection and axios.
// Initialization must start before doing any subscribes to data.
init({ baseUrl: BASE_URL, orm })

// Global initialization logic
function useGlobalInit (session) {
  let $session = useModel('_session')

  // Initialize _session on mobile
  useMemo(() => session && $session.setEach(session), [])

  let [userId] = useSession('userId')
  let [user, $user] = useDoc('users', userId)

  // Create an anonymous user data if it doesn't exist.
  // You would want to remove this logic after adding real authorization.
  if (!user) {
    console.warn('Anonymous user detected. Creating DUMMY user data...')
    throw $user.createAsync()
  }

  useMemo(() => {
    // reference self to '_session.user' for easier access
    $session.ref('user', $user)
  }, [])
}

export default observer(({ session }) => {
  useGlobalInit(session)

  return pug`
    TestComponent
  `
})
