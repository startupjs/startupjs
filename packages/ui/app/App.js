import App from '@startupjs/app/client'
export default App
