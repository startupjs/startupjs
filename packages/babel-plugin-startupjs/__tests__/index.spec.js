const pluginTester = require('babel-plugin-tester').default
const plugin = require('../index')
const { name: pluginName } = require('../package.json')

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  pluginOptions: {},
  tests: {
    'Ignores imports of other libs': /* js */`
      import { observer } from 'startupjs'
      import App from 'startupjs/app'
    `,
    'Transforms known import from moduleMap.json': /* js */`
      import { Button } from '@startupjs/ui'
    `,
    'Transforms multiple imports': /* js */`
      import { Button, Span, TextInput, Collapse } from '@startupjs/ui'
    `,
    'Doesn\'t touch unknown imports': /* js */`
      import { Button, UnknownComponent } from '@startupjs/ui'
    `,
    'Doesn\'t touch grep imports': /* js */`
      import * as UI from '@startupjs/ui'
    `,
    'Transforms export': /* js */`
      export { observer } from 'startupjs'
      export { Button, UnknownComponent, Span, TextInput } from '@startupjs/ui'
    `
  }
})
