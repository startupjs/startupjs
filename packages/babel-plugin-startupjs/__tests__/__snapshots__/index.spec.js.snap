// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@startupjs/babel-plugin-startupjs Doesn't touch grep imports: Doesn't touch grep imports 1`] = `

import * as UI from '@startupjs/ui'

      ↓ ↓ ↓ ↓ ↓ ↓

import * as UI from "@startupjs/ui/index";


`;

exports[`@startupjs/babel-plugin-startupjs Doesn't touch unknown imports: Doesn't touch unknown imports 1`] = `

import { Button, UnknownComponent } from '@startupjs/ui'

      ↓ ↓ ↓ ↓ ↓ ↓

import Button from "@startupjs/ui/components/Button";
import { UnknownComponent } from "@startupjs/ui/index";


`;

exports[`@startupjs/babel-plugin-startupjs Ignores imports of other libs: Ignores imports of other libs 1`] = `

import { observer } from 'startupjs'
import App from 'startupjs/app'

      ↓ ↓ ↓ ↓ ↓ ↓

import { observer } from "startupjs";
import App from "startupjs/app";


`;

exports[`@startupjs/babel-plugin-startupjs Transforms export: Transforms export 1`] = `

export { observer } from 'startupjs'
export { Button, UnknownComponent, Span, TextInput } from '@startupjs/ui'

      ↓ ↓ ↓ ↓ ↓ ↓

export { observer } from "startupjs";
export { default as Button } from "@startupjs/ui/components/Button";
export { UnknownComponent } from "@startupjs/ui/index";
export { default as Span } from "@startupjs/ui/components/typography/Span";
export { default as TextInput } from "@startupjs/ui/components/forms/TextInput";


`;

exports[`@startupjs/babel-plugin-startupjs Transforms known import from moduleMap.json: Transforms known import from moduleMap.json 1`] = `

import { Button } from '@startupjs/ui'

      ↓ ↓ ↓ ↓ ↓ ↓

import Button from "@startupjs/ui/components/Button";


`;

exports[`@startupjs/babel-plugin-startupjs Transforms multiple imports: Transforms multiple imports 1`] = `

import { Button, Span, TextInput, Collapse } from '@startupjs/ui'

      ↓ ↓ ↓ ↓ ↓ ↓

import Button from "@startupjs/ui/components/Button";
import Span from "@startupjs/ui/components/typography/Span";
import TextInput from "@startupjs/ui/components/forms/TextInput";
import Collapse from "@startupjs/ui/components/Collapse";


`;
